iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2504,4 +2504,190 @@ TYPED_TEST(stringTyped_test, AccessMaxPositionOfNotEmptyStringViaConstSubscriptO
+    string<10> sut("Hypno");
+    string<10> string_to_insert("toad");
+    ASSERT_TRUE(sut.insert(5, string_to_insert, 4));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Hypnotoad"));
+}
+
+TYPED_TEST(stringTyped_test, InsertCxxStringToEmptyStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "95ce194c-62a7-420c-9f37-9a2d7c5b17b2");
+    string<1> string_to_insert("M");
+    ASSERT_TRUE(this->testSubject.insert(0, string_to_insert, 1));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(string_to_insert.c_str()));
+}
+
+TYPED_TEST(stringTyped_test, InsertEmptyCxxStringDoesNotChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b566efaf-4a32-41a1-970f-4d51675e594f");
+    this->testSubject = "M";
+    string<1> string_to_insert("");
+    ASSERT_TRUE(this->testSubject.insert(0, string_to_insert, 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, InsertCxxStringWithCountGreaterThanSizeOfStringLiteralFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9bd29af2-12c8-4dc1-8ee8-82610f1bb9a6");
+    string<1> string_to_insert("M");
+    ASSERT_FALSE(this->testSubject.insert(0, string_to_insert, 2));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertTooLargeCxxStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e9d88dd2-9da5-4f5f-a798-10b3931e6516");
+    string<10> sut("Ferdinand");
+    string<16> string_to_insert("Spitzschnueffler");
+    ASSERT_FALSE(sut.insert(9, string_to_insert, 16));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Ferdinand"));
+}
+
+TEST(String10, InsertTooLargeCxxStringWithSmallCountSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e99348ac-87a4-4678-b33a-2ddac3e21cb4");
+    string<10> sut("Ferdinand");
+    string<16> string_to_insert("Spitzschnueffler");
+    ASSERT_TRUE(sut.insert(9, string_to_insert, 1));
+    EXPECT_THAT(sut.size(), Eq(10));
+    EXPECT_THAT(sut.c_str(), StrEq("FerdinandS"));
+}
+
+TYPED_TEST(stringTyped_test, InsertCxxStringWithCount0DoesntChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "340c595e-b0b1-43a8-a08e-ebb62dc0d306");
+    string<9> string_to_insert("Ferdinand");
+    ASSERT_TRUE(this->testSubject.insert(0, string_to_insert, 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertCxxStringAtPositionGreaterStringSizeFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a2517ea0-2842-4f56-a89a-b7fd01c6c6ba");
+    string<10> sut("Muesli");
+    string<1> string_to_insert("s");
+    ASSERT_FALSE(sut.insert(7, string_to_insert, 1));
+    EXPECT_THAT(sut.size(), Eq(6));
+    EXPECT_THAT(sut.c_str(), StrEq("Muesli"));
+}
```

Comment:
You added test cases for string literals and `cxx::string`. It would make sense to add the same test cases also for `std::string`, right?

Reply:
The current implementation doesn't allow (via enable_if) to insert std::strings and I tend to keep it like this. @mossmaurice What do you think?

Reply:
I think we cannot use `c_str()` of `std::string` explicitly if we need to insert a string (it does not return a char array). Maybe for the better, so count cannot exceed the size of the inserted string.